src/posts/introducing-inference-data/index.md

```diff
@@ -0,0 +1,122 @@
+The prior predictive distribution, the posterior distribution, the posterior predictive distribution, the log likelihood, and even the input data
+are all numerical quantities that need to be captured and ordered to facilitate analysis.
+
+Often these arrays need to be passed to specialized functions in order to check for Markov Chain Monte Carlo (MCMC) convergence, model comparison, or getting summaries from the target distribution.
+
+And once the individual statistician has completed their work, they may either want to save it to disk,
+or share it with a colleague.
+
+## The Modern PPL designer
+
+Now consider a PPL designer. PPL designers typically want to focus on model building and inference: graph representation, samplers, symbolic differentiation speed and stability.
+However for their PPL to be useful they must ensure that users have easy access to diagnostics,
+visualizations, and model criticism tools.
+Suddenly the scope of their codebase grows, and duplicative functionality now exists across
+the open source Bayesian ecosystem.
+
+## Life before az.InferenceData
+
+Before `az.InferenceData` there wasn't a consistent object or method both for the user or for the PPL library maintainers.
+Things like posteriors, prior predictives, observed data, and sampler stats could be stored in their own ways,
+For example one PPL may store their modeling outputs in a dictionary, other may have used plain NumPy arrays, a third had its own custom class.
+And that's even before considering the shapes and indices of the array.
+
+This all meant
+
+- Each PPL designer would have to implement their own data objects and would have to solve lots problems like how to serialize data to disk.
+- All adjacent workflow libraries would only work with a subset of these objects, or would require users to restructure data each time.
+- The APIs and interfaces for all of these were inconsistent making life challenging for end user. Users of different PPLs would be isolated in completely different ecosystems, even when the PPLs share the same general programming language.
+
+## How az.InferenceData works and its implementation
+
+`az.InferenceData` is an object that stores (nearly) all the outputs of Bayesian Modeling in a consistent manner.
+Under the hood `az.InferenceData` is largely a collection of Xarray objects, with some utility functions built in as well.
+Xarray was the natural choice, because storing indexes, and performing calculations
+over multidimensional outputs are routine tasks for computational Bayesian modelers.
+Named dimensions let us directly interact with the dimensions of interest, such as the quantities being studied.
+Importantly they also remove implicit assumptions of which dimension represents chains and draws,\
+dimensions while necessary for sampling, are not so interesting when trying to perform inference,
+similar to how batch dimensions are a computational nuance, but not a modeling nuance.
```

Comment:
Something seems off here but I'm not sure how to edit it appropriately.

Reply:
I reedited it, let me know if you like how it sounds now